src/http.js

```diff
@@ -161,55 +162,145 @@ export function isFile(obj, navigatorObj) {
     }
     return false
   }
-  if (typeof File !== 'undefined') {
-    // eslint-disable-next-line no-undef
-    return obj instanceof File
+
+  if (typeof File !== 'undefined' && obj instanceof File) { // eslint-disable-line no-undef
+    return true
+  }
+  if (typeof Blob !== 'undefined' && obj instanceof Blob) { // eslint-disable-line no-undef
+    return true
   }
+  if (typeof Buffer !== 'undefined' && obj instanceof Buffer) {
+    return true
+  }
+
   return obj !== null && typeof obj === 'object' && typeof obj.pipe === 'function'
 }
 
-function formatValue(input, skipEncoding) {
-  const {collectionFormat, allowEmptyValue} = input
+function isArrayOfFile(obj, navigatorObj) {
+  return (Array.isArray(obj) && obj.some(v => isFile(v, navigatorObj)))
+}
 
-  // `input` can be string in OAS3 contexts
-  const value = typeof input === 'object' ? input.value : input
+const STYLE_SEPARATORS = {
+  form: ',',
+  spaceDelimited: '%20',
+  pipeDelimited: '|'
+}
 
-  const SEPARATORS = {
-    csv: ',',
-    ssv: '%20',
-    tsv: '%09',
-    pipes: '|'
-  }
+const SEPARATORS = {
+  csv: ',',
+  ssv: '%20',
+  tsv: '%09',
+  pipes: '|'
+}
+
+// Formats a key-value and returns an array of key-value pairs.
+//
+// Return value example 1: [['color', 'blue']]
+// Return value example 2: [['color', 'blue,black,brown']]
+// Return value example 3: [['color', ['blue', 'black', 'brown']]]
+// Return value example 4: [['color', 'R,100,G,200,B,150']]
+// Return value example 5: [['R', '100'], ['G', '200'], ['B', '150']]
+// Return value example 6: [['color[R]', '100'], ['color[G]', '200'], ['color[B]', '150']]
+function formatKeyValue(key, input, skipEncoding = false) {
```

Comment:
Since the key may change depending on the value, `formatValue` has been changed to a method that returns a key-value pair